Improved iOS Game Center functions

romulox-x opened the pull request:
Fixed null pointer bug in ios store. added functions to game center to reset the user's achievements, request all the achievements they've made progress on, request descriptions of all possible achievements, and show the built in game center overlay.  Upgraded all (I think) deprecated functions to ios 6 versions.  as per discussion with reduz on IRC, ios 6 is now the official lowest supported iOS version.  compatibility could be added, but it's so obsolete, it's tough to argue for continuing to support it.  some naming of functions and dictionary parameters might need to change.  I just named them whatever made sense to me :)

romulox-x committed:
Fixed null pointer bug in ios store. added functions to game center to reset the user's achievements, request all the achievements they've made progress on, request descriptions of all possible achievements, and show the built in game center overlay.  Upgraded all (I think) deprecated functions to ios 6 versions.  as per discussion with reduz on IRC, ios 6 is now the official lowest supported iOS version.  compatibility could be added, but it's so obsolete, it's tough to argue for continuing to support it.  some naming of functions and dictionary parameters might need to change.  I just named them whatever made sense to me :)

romulox-x committed:
added data to descriptions dictionary that was missing.  Made the elements that sound weird as plurals consistently singular.  changed int arrays to bool arrays in descriptions

romulox-x committed:
added response event for game center overlay closing

romulox-x commented:
Here's the new wiki page reflecting these changes:
https://copy.com/jvQzW3Yt2VMhOpBO

I couldn't figure out how to fork the wiki so that I could do a PR, so hopefully someone with permission can commit these changes to the wiki master branch.

akien-mga commented:
Just a comment about the commit logs, with git it's usually better to define a commit "title" on the first line, then skip a line, then write your explanations as the "body" of the commit log, e.g.:
```
Add possibility to make French fries

An API to handle fryers inputs as FryerInput event has been added, so that French fries can now be made directly from GDscript. This requires a lot of oil though, and couple introduce regressions in lipophobic games.
```

romulox-x commented:
Fixed.  Well, sort of.  I changed how it looks on GitHub, but I imagine the actual commit log that goes with didn't change.

I'll keep this in mind for future commits.  Thanks for the tip.
